frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -1611,6 +1611,49 @@ function ReaderHighlight:getHighlightBookmarkItem()
     end
 end
 
+function ReaderHighlight:_insertHighlight(page, highlight)
+    local position
+    if self.view.highlight.saved[page] == nil or #self.view.highlight.saved[page] == 0 then
+        position = 1
+    else
+        local page_highlights = self.view.highlight.saved[page]
+        local highlight_beginning = highlight.pos0
+        local highlight_end = highlight.pos1
+        if self.ui.rolling then
+            for i, cur_highlight in ipairs(page_highlights) do
+                local order = self.ui.document:compareXPointers(highlight_beginning, cur_highlight.pos0)
+                if order == 0 then
```

Comment:
Please look at https://github.com/koreader/koreader/pull/8690.

Reply:
Took nil to notice.